apps/accounts/models.py

```diff
@@ -308,3 +312,87 @@ def get_user_id_salt(salt=settings.USER_ID_SALT):
     :return: bytes
     """
     return binascii.unhexlify(salt)
+
+
+class UserPasswordDescriptor(models.Model):
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        editable=False
+    )
+    date = models.DateTimeField(
+        _('Descriptor Created On'),
+        auto_now_add=True,
+        editable=False
+    )
+    salt = models.CharField(
+        verbose_name=_('Salt'),
+        max_length=120,
+        editable=False,
+    )
+    iterations = models.IntegerField(
+        _('Iterations'),
+        default=None,
+        editable=False,
+        blank=True,
+        null=True
+    )
+
+    class Meta:
+        verbose_name = _('Password Descritpor')
+        verbose_name_plural = _('Password Descriptors')
+        unique_together = (("user", "iterations",),)
+        ordering = ['-user', 'iterations', ]
+
+    def create_hash(self, password):
+        # use default password hasher, if not sufficient, can pull in stronger version
+        return PasswordHasher().encode(password, self.salt, self.iterations)
+
+    def _gen_salt(self):
+        self.salt = get_random_string(length=self._meta.get_field('salt').max_length)
+
+    def save(self, *args, **kwargs):
+        if not self.salt:
+            self._gen_salt()
+        if not self.iterations:
+            self.iterations = PasswordHasher.iterations
+        return super(UserPasswordDescriptor, self).save(*args, **kwargs)
+
+    def __str__(self):
+        return '{} [{}]'.format(self.user, self.iterations)
+
+
+class PastPassword(models.Model):
+    userpassword_desc = models.ForeignKey(
+        UserPasswordDescriptor,
+        on_delete=models.CASCADE,
+        editable=False
+    )
+    password = models.CharField(
+        _('Password Hash'),
+        max_length=255,
+        editable=False
+    )
+    date_created = models.DateTimeField(
+        _('Date Created'),
+        auto_now_add=True,
+        editable=False
+    )
+
+    class Meta:
+        verbose_name = 'Past Password'
+        verbose_name_plural = 'Past Passwords'
+        unique_together = (("userpassword_desc", "password", "date_created"),)
+        ordering = ['-userpassword_desc', 'password', ]
+
+    def __str__(self):
+        return "{} [{}]".format(self.userpassword_desc, self.date_created)
+
+
+class PasswordHasher(PBKDF2PasswordHasher):
+    """
+    We need to keep the old password so that when you update django
+    (or configuration change) hashes have not changed.
+    Therefore, special hasher.
+    """
+    iterations = 20000 * 10
```

Comment:
@jarwilliams1971 @StewGoin Would the iterations value be best placed in to our vault file, or is this not a major concern?  It could be a vault variable similar to the following in the `hhs_oauth_server/settings/base.py` settings file like the one used for the mbi/hicn hashing:
`USER_ID_ITERATIONS = int(env("DJANGO_USER_ID_ITERATIONS", "2"))`

Reply:
@dtisza1 will move password hasher iteration as part of settings:
PASSWORD_HASH_ITERATIONS = int(env("DJANGO_PASSWORD_HASH_ITERATIONS", "200000"))
if the iteration value is from vault, then extra work needed on bluebutton-web-deployment (adding variable for iteration value ...)